Assets/Scripts/pattern/CollisionSubject.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace pattern
+{
+    /// <summary>
+    ///     Abstract CollisionSubject -- behaves like IObservable. Add this to your player or your powerups, probably?
+    ///     TODO: clean up behaviour with collision matrices instead of specifying floor.
+    /// </summary>
+    public abstract class CollisionSubject : MonoBehaviour, IObservable<Collision>
+    {
+        [SerializeField] private GameObject floor;
+        private readonly HashSet<IObserver<Collision>> _observers = new();
+
+
+        private void OnCollisionEnter(Collision other)
+        {
+            if (other.gameObject == floor) return;
```

Comment:
How does equals work in this case? Since floor is a single specific game object, does that mean that using the tile palette & prefabs this condition would only be true for one of the generated objs? Essentially, what is being compared in this equals. For layer-based collision detect it is the layer ID directly.

Reply:
Oh i  just read your todo, disregard previous comment.

Reply:
Hang on, floor right now is still just a gameobject dragged in in the editor - i'll convert this to a draft PR, it shouldn't be needed now that we have layers